src/lightning/pytorch/core/module.py

```diff
@@ -621,8 +621,18 @@ def __check_not_nested(value: dict, name: str) -> None:
     def __check_allowed(v: Any, name: str, value: Any) -> None:
         raise ValueError(f"`self.log({name}, {value})` was called, but `{type(v).__name__}` values cannot be logged")
 
+    @staticmethod
```

Comment:
Can you instead import it from the logger connector?

```dif
diff --git a/tests/tests_pytorch/trainer/logging_/test_train_loop_logging.py b/tests/tests_pytorch/trainer/logging_/test_train_loop_logging.py
index bc2abdacb..2b88c283f 100644
--- a/tests/tests_pytorch/trainer/logging_/test_train_loop_logging.py
+++ b/tests/tests_pytorch/trainer/logging_/test_train_loop_logging.py
@@ -17,11 +17,13 @@ import collections
 import itertools
 from re import escape
 from unittest import mock
-from unittest.mock import call
+from unittest.mock import call, Mock
 
 import numpy as np
 import pytest
 import torch
+
+from lightning.fabric.plugins.precision.utils import _DtypeContextManager
 from lightning.pytorch import Trainer, callbacks
 from lightning.pytorch.callbacks import EarlyStopping, ModelCheckpoint, TQDMProgressBar
 from lightning.pytorch.core.module import LightningModule
@@ -779,3 +781,16 @@ def test_unsqueezed_tensor_logging():
     model.trainer = trainer
     model.log("foo", Tensor([1.2]), on_epoch=True)
     assert trainer.callback_metrics["foo"].ndim == 0
+
+
+def test_log_tensor_wrap_dtype():
+    class MyModel(LightningModule):
```

Reply:
done!